ingestion/src/metadata/ingestion/source/pipeline/dagster.py

```diff
@@ -0,0 +1,174 @@
+            )
+        return cls(config, metadata_config)
+
+    @property
+    def session(self) -> Session:
+        """
+        Return the SQLAlchemy session from the engine
+        """
+        if not self._session:
+            self._session = create_and_bind_session(self.engine)
+
+        return self._session
+
+    def get_run_list(self):
+
+        run_list = self.session.execute("select * from runs")
+        return run_list
+
+    def yield_pipeline(self, pipeline_details) -> Iterable[CreatePipelineRequest]:
+        """
+        Convert a DAG into a Pipeline Entity
+        :param serialized_dag: SerializedDAG from dagster metadata DB
+        :return: Create Pipeline request with tasks
+        """
+        task_list = [{"name": row["pipeline_name"]} for row in self.get_run_list()]
+        dag = pipeline_details
+        yield CreatePipelineRequest(
+            name=pipeline_details["pipeline_name"],
+            description=dag[3],
+            pipelineUrl=pipeline_details[3],  # Just the suffix
+            pipelineLocation=pipeline_details[3],
+            startDate=dag[9],
+            tasks=task_list,
+            service=EntityReference(
+                id=self.context.pipeline_service.id.__root__, type="pipelineService"
+            ),
+        )
+
+    def yield_pipeline_status(self, pipeline_details) -> OMetaPipelineStatus:
+        run_list = self.get_run_list()
+
+        for run in run_list:
+            log_link = f"{self.service_connection.hostPort}/instance/runs/{run.id}"
+
+            task_status = TaskStatus(
+                name=run["pipeline_name"],
+                executionStatus=STATUS_MAP.get(
+                    run["status"].lower(), StatusType.Pending.value
+                ),
+                startTime=datetime_to_ts(run[9]),
+                endTime=datetime_to_ts(run[10]),
+                logLink=log_link,
+            )
+            pipeline_status = PipelineStatus(
+                taskStatus=[task_status],
+                executionStatus=STATUS_MAP.get(
+                    run["status"].lower(), StatusType.Pending.value
+                ),
+                executionDate=run[10].timestamp(),
+            )
+            yield OMetaPipelineStatus(
+                pipeline_fqn=self.context.pipeline.fullyQualifiedName.__root__,
+                pipeline_status=pipeline_status,
+            )
+
+    def yield_pipeline_lineage_details(
+        self, pipeline_details
+    ) -> Optional[Iterable[AddLineageRequest]]:
+        """ """
+
+    def close(self):
+        self.session.close()
+
+    def test_connection(self) -> None:
+        test_connection(self.engine)
+
+    def get_pipelines_list(self):
+
+        result = self.engine.execute(text("SELECT * from runs"))
+        for i in result:
+            yield i
+
+    def get_pipeline_name(self, pipeline_details) -> str:
+        """
+        Get Pipeline Name
+        """
+        return pipeline_details
```

Comment:
Are we retrieving the name correctly from here? I believe this is just the full dict